### PR TITLE
Make structure tree view make scrollable

### LINKED ISSUE #65: When editing timespans, user cannot see both Waveform and inline edit form at the same time
### Description

When editing a timespan with the use of segment handles in the waveform, the user cannot see the respective inline form when the timespan is at the bottom of the tree (even in the middle depending on the screen size of the device the user is using).

### PR DESCRIPTION
**Pull request by Dananji**
Fixes #65

### CODE REVIEW

File: src/components/ButtonSection.js [hunk 1]
```diff
@@ -66,6 +66,9 @@ class ButtonSection extends Component {
   };
 
   handleTimeSpanClick = () => {
+    // Clear existing alertObj
+    this.clearAlert();
+
```
Comment (Dananji) [inline src/components/ButtonSection.js]: When `Add a Timespan` button is clicked when there are no empty time slots to add a new timespan ahead of the current playhead, the application emits a warning. If a user moves the playhead in the waveform and click on `Add a Timespan` button again without closing the alert, it stays while the form collapses for the new timespan.
Adding this here fixes this issue.

File: src/components/HeadingForm.js [hunk 2]
```diff
@@ -46,8 +46,9 @@ class HeadingForm extends Component {
   formIsValid() {
     const { headingTitle } = this.state;
     const titleValid = headingTitle && headingTitle.length > 0;
+    const childOfValid = this.state.headingChildOf.length > 0;
 
-    return titleValid;
+    return titleValid && childOfValid;
```
Comment (Dananji) [inline src/components/HeadingForm.js]: When adding a new heading; after validating the title the `Save` is enabled when `Child Of` option is empty. And if `Save` is clicked without choosing the `Child Of` options it saves the heading to the top of the structure (as 1st child of the root element). This check keeps the `Save` button disabled until the user selects the `Child Of` option.

@adamjarling Was this the expected behavior (before this additional check)?

File: src/components/ButtonSection.js (hunk 1 above)
Comment (adamjarling) [reply src/components/ButtonSection.js]: Good catch.

File: src/components/HeadingForm.js (hunk 2 above)
Comment (adamjarling) [reply src/components/HeadingForm.js]: @Dananji @cjcolvar @bkeese  It depends how the data structure comes down.  Chris or Brian, should a user be able to add another top level element to the data structure, or will all additions be underneath the parent?

Comment (adamjarling) [reply src/components/HeadingForm.js]: top-level-highlight

Comment (Dananji) [reply src/components/HeadingForm.js]: @adamjarling When `Child Of` option is not selected the new heading is added between `utah_phillips_one.222` and `Section 1` without this check. I think we don't have an option to add a heading on top of the existing top element (`utah_phillips_one.222`).
Sorry my previous comment was not clear.

**Pull request merged**